google/cloud/spanner_v1/transaction.py

```diff
@@ -110,7 +110,12 @@ def _execute_request(
         """
         transaction = self._make_txn_selector()
         request.transaction = transaction
-        with trace_call(trace_name, session, attributes):
+        observability_options = getattr(
+            self._session._database, "observability_options", None
```

Comment:
There is nothing called self._session here. This will always return None.
session is an argument to this method.

Please add an argument observability_options to `_execute_request` and infer the options via database and pass it from where ever the `_execute_request` is called.